x/twap/api_test.go

```diff
@@ -140,13 +175,65 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 			input:        makeSimpleTwapInput(baseTime, baseTime.Add(20*time.Second), quoteAssetB),
 			expTwap:      sdk.NewDecWithPrec(15, 2), // .1 for 10s, .2 for 10s
 		},
+		// start at 5 second after first twap record, end at 5 second after second twap record
 		"(2 record) start and end interpolated": {
 			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
 			ctxTime:      baseTime.Add(time.Minute),
 			input:        makeSimpleTwapInput(baseTime.Add(5*time.Second), baseTime.Add(20*time.Second), quoteAssetA),
 			// 10 for 5s, 5 for 10s = 100/15 = 6 + 2/3 = 6.66666666
 			expTwap: ThreePlusOneThird.MulInt64(2),
 		},
+		"(2 record) to now: start time = second record time": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(5), // .1 for 10s, .2 for 10s
+		},
+		"(2 record) to now: first record time < start time < second record time": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(5*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(10), // should only look at record before given start time
+		},
+
+		"(3 record) start and end point to same record": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
```

Comment:
Does `ctxTime` matter here if this is a not "to now" twap? Is it possible to remove it if unused?

Reply:
we still need the ctxTime to be ahead of the record to test `GetArithmeticTwap`